### ISSUE TITLE
use opengraph image as fallback for thumbnails on topic & magazine cards

### ISSUE THREAD

**Issue by SvSven** [equinor/energyvision#1306]
Topic pages and magazine articles using the new video banner won't have a defined hero image; this is a problem when these pages are linked to with cards as we require (and should continue to require) an image.

We can use the open graph image for social media as fallback in these cases.
For the magazine landing page, we will need to add indexing of the open graph image in Algolia similar to how we did with the hero image.

---

## Acceptance criteria
- [ ] Hero image is used as card thumbnail when defined
- [ ] Open graph image is used as card thumbnail if defined and hero image is not defined
- [ ] Card(s) do not render when neither hero image nor open graph image is defined

**Comment by SvSven** [comment]
This is ready for technical review @fernandolucchesi @padms @bhavi25 @nilsml 
It's a pretty simple change, most of the code changes are actually in the search indexer but those are also very simple and straightforward.

We need to do a deployment of the search code for this to work for the magazine index page.

**Comment by fernandolucchesi** [comment]
LGTM ✅

**Comment by SvSven** [comment]
Hi @meols @NickHaggerty1 

With this issue we are adding a change to how cards work with images; because a hero image is not a mandatory field in Sanity, we chose to not render cards for topic pages/magazine articles/news articles if said page does not have a hero image when we first added cards - since it would look very weird.

This issue adds functionality that if there is no hero image, it will check if there is a open graph image for social media defined and use that as fallback. The main idea behind it was that the new video banner won't have a hero image, so we needed a different way of handling that - but we're currently also looking into using the thumbnail functionality of Mux for that.

In reality this probably won't have much of an impact as the editorial team is pretty good with always adding a hero image, but it's good to have this as a "just in case" feature.
